src/components/WeatherApp.jsx

```diff
@@ -1,56 +1,110 @@
-import React, { useEffect } from 'react'
+import React, { useEffect, useState } from 'react'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faDroplet } from '@fortawesome/free-solid-svg-icons'
+import BeatLoader from "react-spinners/BeatLoader";
+import { ToastContainer, toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
 import { FiSearch } from 'react-icons/fi'
-import { BsDropletFill } from 'react-icons/bs'
 import { FaWind } from 'react-icons/fa'
 const WeatherApp = () => {
+    const [weatherData, setWeatherData] = useState({})
+    const [isLoading, setIsLoading] = useState(false)
+    const [inputName, setInputName] = useState("Pune");
+    const [saveInputText, setSaveInputText] = useState("Pune");
+    const [error, setError] = useState({ errorText: "", showError: "removeErrorSymbol" })
 
     useEffect(() => {
-        fetch("http://api.weatherapi.com/v1/current.json?key=5bca6bccf3ce4c41a8f15340232807&q=Pune")
+        setIsLoading(true);
+        fetch(`https://api.weatherapi.com/v1/current.json?key=5bca6bccf3ce4c41a8f15340232807&q=${saveInputText}`)
             .then((response) => response.json())
-            .then((data) => console.log(data))
+            .then((data) => {
+                setIsLoading(false)
+                if (data.error) {
+                    setWeatherData(data)
+                    setError({ errorText: "?", showError: "displayErrorSymbol" })
+                    toast.error("Location in valid , please enter a city name.")
+                }
+                else {
+                    setWeatherData(data);
+                    setError({ errorText: "", showError: "removeErrorSymbol" })
+                }
+
+            })
             .catch((error) => console.log(error))
-    }, [])
+    }, [saveInputText])
+    const checkOnKeyDown = (e) => {
```

Comment:
lets use function naming like this : `handle`+`FunctionPurpose`
so above function will be named `handleInputBoxKeydown`